cpp/src/dbscan/runner.cuh

```diff
@@ -173,9 +173,9 @@ size_t run(const ML::cumlHandle_impl& handle, Type_f* x, Index_ N, Index_ D,
       adj_graph.resize(adjlen, stream);
```

Comment:
If we are using `curradjlen` below, then this `resize` method should also be using `curradjlen`, right?

Reply:
This was actually a bit confusing for me as well. Essentially, `adjlen` is actually the maximal sparse adjacency matrix length (so maybe we should rename it `maxAdjlen`?). And it does make sense not to `resize` in later batches with smaller size as this saves some memory ops, as the adj-graph alogs properly over-write this space, even if it's too large.
The only inconvenience right now is that if there is a bug in adj-graph (or in the weak_cc kernel processing it) then cuda-memcheck cannot always find it.

Reply:
>  The only inconvenience right now is that if there is a bug in adj-graph (or in the weak_cc kernel processing it) then cuda-memcheck cannot always find it.

Ideally, I'd expect running cuda-memcheck on unit-tests for dbscan to be able to catch these.

Reply:
Agreed. I think it is better to rename this variable to `maxadjlen`.

Reply:
I have changed this.